x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/transforms/TransformHealthIssue.java

```diff
@@ -43,12 +47,21 @@ public TransformHealthIssue(String issue, String details, int count, Instant fir
     }
 
     public TransformHealthIssue(StreamInput in) throws IOException {
+        if (in.getTransportVersion().onOrAfter(TransportVersion.V_8_8_0)) {
+            this.type = in.readString();
+        } else {
+            this.type = null;
```

Comment:
The `readString()` and `writeString()` methods used in this class assume `type` is not `null`.

You'll need to either assign a default value here, like `pre_8.8`, or else use `readOptionalString()`/`writeOptionalString()`.

Reply:
Done (using default value).

Initially I thought this is not needed, because `TransformHealthIssue` is never persisted, but after your comment I realized it can be indeed a problem while we upgrade the cluster to `8.8` and the request is being sent between nodes with different versions.